python/paddle/fluid/regularizer.py

```diff
@@ -231,6 +263,95 @@ def __str__(self):
         return "L1Decay, regularization_coeff=%f" % self._regularization_coeff
 
 
+class WeightDecayBase(object):
+    def __str__(self):
+        raise NotImplementedError()
+
+    def __call__(self, params_and_grads):
+        raise NotImplementedError()
+
+    def apply(self):
+        raise NotImplementedError()
+
+
+class WeightDecay(WeightDecayBase):
+    """
+    WeightDecay is used to update the optimized parameters by using the
+    parameters before optimization.
+
+    Args:
+        coeff (float|Variable): The maximum value to clip by.
+        attempt_decay_param_fun (function|None): If it is not None,
+            only variables that makes attempt_decay_param_fun(variable)==True
+            will be updated. It only works when we want to specify variables.
+            Default: None.
+
+    Examples:
+        .. code-block:: python
+            learning_rate = 0.1
+            optimizer = fluid.optimizer.Adagrad(
+                learning_rate=learning_rate,
+                weight_decay=fluid.weight_decay.WeightDecay(
+                    coeff=learning_rate))
+    """
+
+    def __init__(self, coeff=0.0, attempt_decay_param_fun=None):
+
+        if not isinstance(coeff, float) and \
+                not isinstance(coeff, framework.Variable):
+            raise TypeError("coeff should be float or Variable.")
+
+        self.scaled_params_ = dict()
+        self.params_name_ = []
+        self.attempt_decay_param_fun_ = attempt_decay_param_fun
+        self.coeff_ = coeff
+
+    def __call__(self, params_and_grads):
+        from . import layers
+        if self.coeff_ == 0.0:
```

Comment:
There are two problems of this line:

1. Should we use `==` when coeff_ is a `float` value?
2. When `self.coeff_` is a variable, `==` is overwrited by generating an equal operator.